src/main/java/graphql/language/ImplementingTypeDefinition.java

```diff
@@ -0,0 +1,19 @@
+package graphql.language;
+
+
+import graphql.PublicApi;
+
+import java.util.List;
+
+/**
+ * A {@link TypeDefinition} that might implement interfaces
+ *
+ * @param <T>
+ */
+@PublicApi
+public interface ImplementingTypeDefinition<T extends ImplementingTypeDefinition> extends TypeDefinition<T> {
```

Comment:
why the recursive type definition here?

Reply:
Yep, not needed. `<T extends TypeDefinition>` should be enough.